xclim/indices/_threshold.py

```diff
@@ -1237,6 +1238,62 @@ def days_with_snow(
     return to_agg_units(out, prsn, "count")
 
 
+@declare_units(tasmax="[temperature]", hurs="[g m-3]", thresh="[temperature]")
+def heat_index(
+    tasmax: xarray.DataArray,
+    hurs: xarray.DataArray,
+    thresh: str = "20.0 degC",
+    freq: str = "MS",
+) -> xarray.DataArray:
+    r"""Daily heat index.
+
+    Days in which the human body perceives temperature is hot, which combines
+    relative humidity with the air temperature.
+
+    Parameters
+    ----------
+    tasmax : xarray.DataArray
+      Maximum daily temperature.
+    hurs : xarray.DataArray
+      Relative humidity.
+    thresh : str
+      Threshold temperature on which to base evaluation.
+    freq : str
+      Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray, [time][temperature]
+      Heat index.
+    """
+    thresh = convert_units_to(thresh, tasmax)
+    t = convert_units_to(tasmax, "degC")
+    t = t.where(t > 20)
```

Comment:
```suggestion
    thresh = convert_units_to(thresh, "degC")
    t = convert_units_to(tasmax, "degC")
    t = t.where(t > thresh)
```
I'm guessing that hardcoded 20 was forgotten? Also, better to convert everything _explicitly_ to the same units if we need a specific one.